backend/backend/app.py

```diff
@@ -1,8 +1,73 @@
 from flask import Flask
+from flask_restful import Api, Resource, reqparse
+from flask_cors import CORS
+import backend.calculator as calculator
+from flask_pymongo import PyMongo
 
 app = Flask(__name__)
+api = Api(app)
+cors = CORS(app, resources={r"/mortgageCalculator/*": {"origins": "*"}})
+mongodb_client = PyMongo(app, uri="mongodb://localhost:27017/calculator_data")
+db = mongodb_client.db
 
+mortgageCalculator_put_args = reqparse.RequestParser()
```

Comment:
Some general comment about the validation for parameters here:

Right now we validate only values of each one of params, but what about following scenarios:

user posts data like this:
```json
"interest_type": "WIBOR",
"installment_type": "fixed",
"down_payment": 0,
"credit_amount": 1000,
"loan_term": 0,
"interest_rate": 0, 
"commission": 0,
```

or like this:
```json
"interest_type": "fixed",
"installment_type": "descending",
"down_payment": 0,
"credit_amount": 1000,
"loan_term": 0,
"interest_rate": 0, 
"commission": 0,
```

Further validation of those parameters is required, as if you put some garbage data into backend it will crash catastrophically
For sure some values need to be validated for minimal value (interest_rate, loan_term, credit_amount) 
Mentioned variables cannot have a 0 or negative integers values 

`commission` and `down_payment` - I guess we don't need those values right now?

`installment_type` and `interest_type` have a cross value binding and if user will select them wrongly we should raise some warning (but not crash whole server application)

Reply:
Validation for `int` values added as suggested,
Validation for bounded keys `interest_type` and `installment_type` added as a conditional statement,
Lastly `commission` and `down_payment` still present but they will be used in next iteration, thus they stay as they are 

Resolving topic